### PR TITLE
alex/refactor pod cli

### PR DESCRIPTION
**Pull request by wadealexc**

### CODE REVIEW

File: cli/core/utils.go [hunk 1]
```diff
@@ -272,42 +272,67 @@ func WriteOutputToFileOrStdout(output []byte, out *string) {
 	}
 }
 
-func FilterNotCheckpointedOrWithdrawnValidators(
-	allValidatorsForEigenpod []ValidatorWithIndex,
-	onchainInfo []onchain.IEigenPodValidatorInfo,
+func SelectCheckpointableValidators(
+	client *ethclient.Client,
+	eigenpodAddress string,
+	validators []ValidatorWithIndex,
```
Comment (jbrower95) [inline cli/core/utils.go:278]: hmm, is this better?

it's a bit simpler to have this function be stateless and literally just a filter, than to have it data-fetch as well

e.g if we end up calling two of these throughout a command by accident we're going to data fetch twice, and those RPC calls get rate-limited 

if you feel passionately about this sure but idk

Comment (wadealexc) [reply cli/core/utils.go:278]: The things I like better about this:
1. The naming is a lot more clear
2. All the select/filter methods behave the same way

I hear you on this:

> e.g if we end up calling two of these throughout a command by accident we're going to data fetch twice, and those RPC calls get rate-limited

But I specifically looked at the places these were getting called and this behaves identically to the previous implementation. If we change things in the future we might want to change how this works, but for now - this felt fine

File: cli/core/utils.go [hunk 2]
```diff
@@ -272,42 +272,67 @@ func WriteOutputToFileOrStdout(output []byte, out *string) {
...
 
-func FilterInactiveValidators(
-	allValidatorsForEigenpod []ValidatorWithIndex,
-	onchainInfo []onchain.IEigenPodValidatorInfo,
-) []uint64 {
-	var checkpointValidatorIndices = []uint64{}
-	for i := 0; i < len(allValidatorsForEigenpod); i++ {
-		validator := allValidatorsForEigenpod[i]
-		validatorInfo := onchainInfo[i]
+func SelectAwaitingCredentialValidators(
```
Comment (jbrower95) [inline cli/core/utils.go:301]: replace literally all this code with SelectValidatorsWhere() and pass in a lambda for this bit (`validatorInfo.Status == ValidatorStatusInactive) && (validator.Validator.ExitEpoch == FAR_FUTURE_EPOCH`)

Comment (wadealexc) [reply cli/core/utils.go:301]: i could pass in a lambda for 2/3 but the top function `SelectCheckpointable...` has an extra term

I guess I havent used lambdas in go in a while - can I do that cleanly?

**Review by jbrower95 — APPROVED** [review]

**Pull request merged**